src/abstract/ZoweTreeProvider.ts

```diff
@@ -126,16 +126,15 @@ export class ZoweTreeProvider {
         return undefined;
     }
 
-    public async editSession(node: IZoweDatasetTreeNode) {
+    public async editSession(node: IZoweTreeNode) {
         const profile = node.getProfile();
         const profileName = node.getProfileName();
         const EditSession = await Profiles.getInstance().editSession(profile, profileName);
-
         if (EditSession) {
             node.getProfile().profile= EditSession as IProfile;
-            setProfile(node, EditSession as IProfile);
-            setSession(node, EditSession as ISession);
-            this.refresh();
+            await setProfile(node, EditSession as IProfile);
+            await setSession(node, EditSession as ISession);
+            await this.refresh();
```

Comment:
Does the `await` affect the `this.refresh()`?

Reply:
yes, as far as I could see. It will wait for setProfile and setSession before refreshing. If I remove the await it will execute first and not wait for the other functions. which is something that I do not want.

Reply:
I see, the reason I asked was because to my understanding, the `this.refresh()` isn't an `async` function so it doesn't return a `Promise`, so I don't think the `await` is needed and it won't affect the order of execution.

Reply:
removed